src/parse/patterns/pattern.js

```diff
@@ -17,14 +36,16 @@ function patternName (patternFile) {
  * Generate a pattern entry (object) by extending the patternFile
  * with a few things.
  */
-function parsePattern (patternFile, patternData, relativeRoot) {
-  return Object.assign(
+function parsePattern (patternFile, patternData, relativeRoot, options) {
+  let parsedPattern = Object.assign(
     patternFile,
     {
       id: resourceId(patternFile, relativeRoot, 'patterns'),
       name: patternName(patternFile)
     }
   );
+  parsedPattern = parseDataFields(parsedPattern, options);
```

Comment:
Yeah, it just felt a bit deeply nested/hard to read. I can give it another whack.